### ISSUE TITLE
SC3016 and SC3017 are not issued for `test ! -a` and `test ! -v`, respectively

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3173]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC3016 and SC3017
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
test ! -a /bin/sh
[ ! -a /bin/sh ]
test ! -v SHELL
[ ! -v SHELL ]
```

#### Here's what shellcheck currently says:

```
Line 3	SC3017: In POSIX sh, unary -a in place of -e is undefined.
Line 5	SC3016: In POSIX sh, unary -v (in place of [ -n "${var+x}" ]) is undefined.
```

#### Here's what I wanted or expected to see:

```
Line 2	SC3017: In POSIX sh, unary -a in place of -e is undefined.
Line 3	SC3017: In POSIX sh, unary -a in place of -e is undefined.
Line 4	SC3016: In POSIX sh, unary -v (in place of [ -n "${var+x}" ]) is undefined.
Line 5	SC3016: In POSIX sh, unary -v (in place of [ -n "${var+x}" ]) is undefined.
```
